test/unit/EventAccumulator.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { expect, sinon } from '../testUtils.js'
+import Countly from 'countly-sdk-nodejs'
+import { EventAccumulator } from '../../src/EventAccumulator.js'
+
+const sleep = async (ms: number): Promise<unknown> => await new Promise(resolve => setTimeout(resolve, ms))
+const sandbox = sinon.createSandbox()
+const addEventListenerStub = sandbox.stub()
+globalThis.addEventListener = addEventListenerStub
+
+describe('EventAccumulator', () => {
+  let countlyStub: sinon.SinonStubbedInstance<typeof Countly>
+
+  beforeEach(() => {
+    countlyStub = sandbox.stub(Countly)
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it('should accumulate events', async () => {
+    const accumulator = new EventAccumulator(countlyStub)
+    const event = {
+      key: 'test',
+      count: 1,
+      sum: 1,
+      segmentation: {}
+    }
+    // add the event twice with a 100ms delay between them, this is needed to test the duration.
+    accumulator.addEvent(event)
+    await sleep(100)
```

Comment:
https://sinonjs.org/releases/v15/fake-timers/
```suggestion
    clock.tick(100)
```